x/permission/types/common.go

```diff
@@ -12,13 +14,20 @@ func NewPrincipalWithAccount(addr sdk.AccAddress) *Principal {
 	}
 }
 
-func NewPrincipalWithGroup(groupID sdkmath.Uint) *Principal {
+func NewPrincipalWithGroupId(groupID sdkmath.Uint) *Principal {
 	return &Principal{
 		Type:  PRINCIPAL_TYPE_GNFD_GROUP,
 		Value: groupID.String(),
 	}
 }
 
+func NewPrincipalWithGroupInfo(groupOwner sdk.AccAddress, groupName string) *Principal {
+	return &Principal{
+		Type:  PRINCIPAL_TYPE_GNFD_GROUP,
+		Value: types.NewGroupGRN(groupOwner, groupName).String(),
```

Comment:
Need to update validateBasic to allow this format.

```
func (p *Principal) ValidateBasic() error {
	switch p.Type {
	case PRINCIPAL_TYPE_UNSPECIFIED:
		return ErrInvalidPrincipal.Wrapf("Not allowed empty principal type.")
	case PRINCIPAL_TYPE_GNFD_ACCOUNT:
		_, err := sdk.AccAddressFromHexUnsafe(p.Value)
		if err != nil {
			return ErrInvalidPrincipal.Wrapf("Invalid account, principal: %s, err: %s", p.String(), err)
		}
	case PRINCIPAL_TYPE_GNFD_GROUP:
		groupID, err := sdkmath.ParseUint(p.Value)
		if err != nil {
			return ErrInvalidPrincipal.Wrapf("Invalid groupID, principal: %s, err: %s", p.String(), err)
		}
		if groupID.Equal(sdkmath.ZeroUint()) {
			return ErrInvalidPrincipal.Wrapf("Zero groupID, principal %s", p.String())
		}
	default:
		return ErrInvalidPrincipal.Wrapf("Unknown principal type.")
	}
	return nil
}
```

Reply:
fixed.